src/vs/platform/backup/electron-main/backupMainService.ts

```diff
@@ -103,182 +112,222 @@ export class BackupMainService implements IBackupMainService {
+			this.emptyWorkspaces.push(backupFolder);
+			this.saveSync();
 		}
-
-		const sanitizedWorkspaceIdentifier = this.sanitizeId(workspaceIdentifier);
-
-		return arrays.firstIndex(target, id => this.sanitizeId(id) === sanitizedWorkspaceIdentifier);
+		return this.getBackupPath(backupFolder);
 	}
 
-	private sanitizeId(workspaceIdentifier: IWorkspaceIdentifier | ISingleFolderWorkspaceIdentifier): string {
-		if (isSingleFolderWorkspaceIdentifier(workspaceIdentifier)) {
-			return this.sanitizePath(workspaceIdentifier);
+	public unregisterEmptyWindowBackupSync(backupFolder: string): void {
+		let index = arrays.firstIndex(this.emptyWorkspaces, w => isEqual(w, backupFolder, !platform.isLinux));
+		if (index !== -1) {
+			this.emptyWorkspaces.splice(index, 1);
+			this.saveSync();
 		}
-
-		return workspaceIdentifier.id;
 	}
 
 	protected loadSync(): void {
+
 		let backups: IBackupWorkspacesFormat;
 		try {
 			backups = JSON.parse(fs.readFileSync(this.workspacesJsonPath, 'utf8').toString()); // invalid JSON or permission issue can happen here
 		} catch (error) {
 			backups = Object.create(null);
 		}
 
-		// Ensure rootWorkspaces is a object[]
-		if (backups.rootWorkspaces) {
-			const rws = backups.rootWorkspaces;
-			if (!Array.isArray(rws) || rws.some(r => typeof r !== 'object')) {
-				backups.rootWorkspaces = [];
-			}
-		} else {
-			backups.rootWorkspaces = [];
-		}
+		// read empty worrkspace backs first
+		this.emptyWorkspaces = this.validateEmptyWorkspaces(backups.emptyWorkspaces);
 
-		// Ensure folderWorkspaces is a string[]
-		if (backups.folderWorkspaces) {
-			const fws = backups.folderWorkspaces;
-			if (!Array.isArray(fws) || fws.some(f => typeof f !== 'string')) {
-				backups.folderWorkspaces = [];
-			}
-		} else {
-			backups.folderWorkspaces = [];
-		}
+		// read workspace backups
+		this.rootWorkspaces = this.validateWorkspaces(backups.rootWorkspaces);
 
-		// Ensure emptyWorkspaces is a string[]
-		if (backups.emptyWorkspaces) {
-			const fws = backups.emptyWorkspaces;
-			if (!Array.isArray(fws) || fws.some(f => typeof f !== 'string')) {
-				backups.emptyWorkspaces = [];
+		// read folder backups
+		let workspaceFolders: URI[];
+		try {
+			if (Array.isArray(backups.folderURIWorkspaces)) {
+				workspaceFolders = backups.folderURIWorkspaces.map(f => URI.parse(f));
+			} else if (Array.isArray(backups.folderWorkspaces)) {
+				// migrate legacy folder paths
+				workspaceFolders = [];
+				for (const folderPath of backups.folderWorkspaces) {
+					const oldFolderHash = this.getLegacyFolderHash(folderPath);
+					const folderUri = URI.file(folderPath);
+					const newFolderHash = this.getFolderHash(folderUri);
+					if (newFolderHash !== oldFolderHash) {
+						this.moveBackupFolderSync(this.getBackupPath(newFolderHash), this.getBackupPath(oldFolderHash));
+					}
+					workspaceFolders.push(folderUri);
+				}
 			}
-		} else {
-			backups.emptyWorkspaces = [];
+		} catch (e) {
+			// ignore URI parsing exceptions
```

Comment:
Can the scope of this try catch be reduced? A lot of functions are called in here and unexpected errors could be getting captured?

Reply:
@aeschli Can you please address @Tyriar comments?